### ISSUE TITLE
TypeError: Path must be a string. Received undefined

### ISSUE THREAD

**Issue by evilebottnawi** [kristerkari/stylelint-scss#54]
In rule `partial-no-import` https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/partial-no-import/index.js#L18. If i run 
```javascript
stylelint.lint({
        code: someScss,
        syntax: 'scss',
        config
    })
```
I have no `root.source.input.file`

**Comment by dryoma** [comment]
Hmm, I suppose some kind of a warning message is needed for cases like that.

**Comment by dryoma** [comment]
@evilebottnawi is that an urgent matter? I am asking because publishing patches messes up our git history a bit, and this rule is not designed to work without an *actual* file anyway. I made a commit to prevent it from throwing, yet in a Node API code like the one above the rule won't work. Maybe you could disable it for now?

**Comment by evilebottnawi** [comment]
@dryoma can, maybe ignore this rule (early return from rules) if file not present?

**Comment by dryoma** [comment]
Yep, that's what I did (but instead of ignoring there is a stylelint warning message).

**Comment by evilebottnawi** [comment]
@dryoma we don't have similar rules
/cc @jeddy3 @davidtheclark what do you think guys?

**Comment by jeddy3** [comment]
I think leave it down to the user to resolve this by turning off this rule if they aren't working with files.
